internal/validations/clusterlogforwarder/inputs/validate.go

```diff
@@ -49,21 +49,17 @@ func Verify(inputs []loggingv1.InputSpec, status *loggingv1.ClusterLogForwarderS
 		case !validAudit(input, status, extras):
 		case input.Receiver != nil && !extras[constants.VectorName]:
 			badInput("ReceiverSpecs are only supported for the vector log collector")
-		case input.Receiver != nil && input.Receiver.ReceiverTypeSpec == nil:
-			badInput("invalid ReceiverTypeSpec specified for receiver")
 		case input.Receiver != nil && input.Receiver.Type != loggingv1.ReceiverTypeHttp && input.Receiver.Type != loggingv1.ReceiverTypeSyslog:
 			badInput("invalid Type specified for receiver")
-		case input.Receiver != nil && input.Receiver.Type == loggingv1.ReceiverTypeHttp && input.Receiver.Syslog != nil:
+		case input.Receiver != nil && input.Receiver.Type == loggingv1.ReceiverTypeHttp && input.Receiver.ReceiverTypeSpec != nil && input.Receiver.Syslog != nil:
```

Comment:
Can you use the above `isReceiverSpecDefined()` to simplify it a bit along with `isHttpReceiver()`?
```suggestion
		case isReceiverSpecDefined(input) && input.Receiver.IsHttpReceiver() && input.Receiver.Syslog != nil:
```